executor/tests/test_progress.py

```diff
@@ -1,61 +1,58 @@
+import errno
 import json
 import logging
 import math
-import re
 import time
 import unittest
 from threading import Event, Thread
 
-import os
-from nose.tools import *
-
 import cook.executor as ce
 import cook.progress as cp
-from tests.utils import assert_message, ensure_directory, get_random_task_id, parse_message, FakeMesosExecutorDriver
+import tests.utils as tu
 
 
 class ProgressTest(unittest.TestCase):
     def test_match_progress_update(self):
 
         progress_regex_string = '\^\^\^\^JOB-PROGRESS:\s+([0-9]*\.?[0-9]+)($|\s+.*)'
-        progress_regex_pattern = re.compile(progress_regex_string)
+        progress_watcher = cp.ProgressWatcher('', '', None, 1, progress_regex_string, None, None, None)
 
         def match_progress_update(input_string):
-            return cp.ProgressWatcher.match_progress_update(progress_regex_pattern, input_string)
-
-        self.assertIsNone(match_progress_update("One percent complete"))
-        self.assertIsNone(match_progress_update("^^^^JOB-PROGRESS: 1done"))
-        self.assertIsNone(match_progress_update("^^^^JOB-PROGRESS: 1.0done"))
-        self.assertIsNone(match_progress_update("^^^^JOB-PROGRESS 1 One percent complete"))
-        self.assertIsNone(match_progress_update("JOB-PROGRESS: 1 One percent complete"))
-
-        self.assertEqual(('1', ''),
-                         match_progress_update("^^^^JOB-PROGRESS: 1"))
-        self.assertEqual(('1', ''),
-                         match_progress_update("^^^^JOB-PROGRESS:   1"))
-        self.assertEqual(('1', '    '),
-                         match_progress_update("^^^^JOB-PROGRESS:   1    "))
-        self.assertEqual(('1', ' done'),
-                         match_progress_update("^^^^JOB-PROGRESS:   1 done"))
-        self.assertEqual(('1', ' One percent complete'),
-                         match_progress_update("^^^^JOB-PROGRESS: 1 One percent complete"))
-        self.assertEqual(('1', '    One percent complete'),
-                         match_progress_update("^^^^JOB-PROGRESS: 1    One percent complete"))
-        self.assertEqual(('50', ' Fifty percent complete'),
-                         match_progress_update("^^^^JOB-PROGRESS: 50 Fifty percent complete"))
+            return progress_watcher.match_progress_update(input_string)
+
+        self.assertIsNone(match_progress_update(b'One percent complete'))
```

Comment:
Why did these change to byte strings?

Reply:
I changed them to match what we expect while executing the code (byte strings from the `tail` function).